sycl/test/abi/sycl_symbols_linux.dump

```diff
@@ -4338,7 +4338,6 @@ _ZNK4sycl3_V16device8get_infoINS0_4info6device32atomic_memory_order_capabilities
 _ZNK4sycl3_V16device8get_infoINS0_4info6device32atomic_memory_scope_capabilitiesEEENS0_6detail19is_device_info_descIT_E11return_typeEv
 _ZNK4sycl3_V16device8get_infoINS0_4info6device33ext_intel_gpu_subslices_per_sliceEEENS0_6detail19is_device_info_descIT_E11return_typeEv
 _ZNK4sycl3_V16device8get_infoINS0_4info6device33ext_oneapi_max_global_work_groupsEEENS0_6detail19is_device_info_descIT_E11return_typeEv
-_ZNK4sycl3_V16device8get_infoINS0_4info6device33usm_restricted_shared_allocationsEEENS0_6detail19is_device_info_descIT_E11return_typeEv
```

Comment:
We can't remove symbols, because that would be an ABI break and currently there is no window open for that, see [ABIPolicyGuide](https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ABIPolicyGuide.md#changing-abi). Instead, you should add a `TODO`/`FIXME` comment saying "remove this API when ABI break is allowed

Reply:
I see. Once these are re-added to the `*.dump` files, the corresponding `sycl/test/abi/*` tests fail as the actual code is missing https://github.com/intel/llvm/pull/8751/files/b7613811cb4b999efee9188f91e4791c23139722#diff-8c94ce4df236f696ac1856ea30578b47d972a9ffa561c18f256567cddc431e5e.

I suppose the only fix is to reintroduce the code and simply deprecate the aspect? I don't think removing it is now an option due to ABI breaking when the `get_info` implementation goes. But it would be odd to have it in the implementation when it is not a part of the SYCL 2020 specification at all.

Reply:
> I suppose the only fix is to reintroduce the code and simply deprecate the aspect?

Yeah, I guess so.

> But it would be odd to have it in the implementation when it is not a part of the SYCL 2020 specification at all.

Well, we do have some outdated/legacy/non-existing stuff, so it wouldn't be extra weird, it happens.